src/transformers/models/zoedepth/modeling_zoedepth.py

```diff
@@ -0,0 +1,1349 @@
+        if self.bin_centers_type == "normed" or self.bin_centers_type == "hybrid2":
+            prev_bin = (seed_bin_centers - min_depth) / (max_depth - min_depth)
+        else:
+            prev_bin = seed_bin_centers
+        prev_bin_embedding = self.seed_projector(x)
+
+        attractors = self.attractors[bin_configurations_name]
+        for projector, attractor, feature in zip(self.projectors, attractors, feature_blocks):
+            bin_embedding = projector(feature)
+            bin, bin_centers = attractor(bin_embedding, prev_bin, prev_bin_embedding, interpolate=True)
+            prev_bin = bin
+            prev_bin_embedding = bin_embedding
+
+        last = outconv_activation
+
+        bin_centers = nn.functional.interpolate(bin_centers, last.shape[-2:], mode="bilinear", align_corners=True)
+        bin_embedding = nn.functional.interpolate(bin_embedding, last.shape[-2:], mode="bilinear", align_corners=True)
+
+        conditional_log_binomial = self.conditional_log_binomial[bin_configurations_name]
+        x = conditional_log_binomial(last, bin_embedding)
+
+        # Now depth value is Sum px * cx , where cx are bin_centers from the last bin tensor
+        out = torch.sum(x * bin_centers, dim=1, keepdim=True)
+
+        return out, domain_logits
+
+
+class ZoeDepthMetricDepthEstimationHead(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+        n_bins = config.n_bins
+        bin_embedding_dim = config.bin_embedding_dim
+        min_depth = config.min_depth
+        max_depth = config.max_depth
+        n_attractors = config.num_attractors
+        attractor_alpha = config.attractor_alpha
+        attractor_gamma = config.attractor_gamma
+        attractor_kind = config.attractor_kind
+        min_temp = config.min_temp
+        max_temp = config.max_temp
+        bin_centers_type = config.bin_centers_type
+
+        self.min_depth = min_depth
+        self.max_depth = max_depth
+        self.bin_centers_type = bin_centers_type
+
+        # Bottleneck convolution
+        bottleneck_features = config.bottleneck_features
+        self.conv2 = nn.Conv2d(bottleneck_features, bottleneck_features, kernel_size=1, stride=1, padding=0)
+
+        # Regressor and attractor
+        if self.bin_centers_type == "normed":
+            SeedBinRegressorLayer = ZoeDepthSeedBinRegressor
+            Attractor = ZoeDepthAttractorLayer
+        elif self.bin_centers_type == "softplus":
+            SeedBinRegressorLayer = ZoeDepthSeedBinRegressorUnnormed
+            Attractor = ZoeDepthAttractorLayerUnnormed
+
+        self.seed_bin_regressor = SeedBinRegressorLayer(
+            bottleneck_features, n_bins=n_bins, min_depth=min_depth, max_depth=max_depth
+        )
+        self.seed_projector = ZoeDepthProjector(bottleneck_features, bin_embedding_dim)
+
+        self.projectors = nn.ModuleList(
+            [ZoeDepthProjector(config.fusion_hidden_size, bin_embedding_dim) for _ in range(4)]
```

Comment:
Same reply here as above, this is different depending on whether a single metric head or multiple are used.

Reply:
In the case of multiple heads, then the `mlp_dim` of `ZoeDepthProjector` is set to `config.bin_embedding_dim // 2` (which is 128 // 2 = 64), whereas in case only a single head is used, then the `mlp_dim` defaults to 128.